tests/cli/validation.test.js

```diff
@@ -9,6 +9,206 @@ describe('Validation', () => {
       exitCode: 1,
     },
   )
+  cliTest(
```

Comment:
I'm going to skip reviewing the tests for now, because they may change a little after the code review has been addressed. I'm glad to see so many tests though!

Reply:
Tests have been updated.